src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -91,18 +92,37 @@ public function __construct(ParameterBagInterface $parameterBag = null)
      */
     public function compile()
     {
-        $this->parameterBag->resolve();
+        if ($this->isCompiled()) {
+            return;
+        }
 
+        $this->parameterBag->resolve();
         $this->parameterBag = new FrozenParameterBag($this->parameterBag->all());
+
+        $this->compiled = true;
```

Comment:
I'd keep only this line a revert the previous changes in this method, that would prevent any potential BC-related side effect of the change.